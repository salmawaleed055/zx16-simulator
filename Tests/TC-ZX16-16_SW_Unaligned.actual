Loaded 10 bytes into memory
0x0000: 0271    xori x1, 1
0x0002: 21B9    li a0, 16
0x0004: 23B9    li a0, 17
0x0006: F2A3    unknown S-type
0x0008: FFC7    Unknown opcode

--- Final State ---
x0: 0x0000
x1: 0x0001
x2: 0x0000
x3: 0x0000
x4: 0x0000
x5: 0x0000
a0: 0x0011
a1: 0x0000
PC: 0x0008
---------------------

Simulation finished.
