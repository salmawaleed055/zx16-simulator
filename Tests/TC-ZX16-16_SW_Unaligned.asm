# TC-ZX16-16_SW_Unaligned.s
# Test Objective: Validate SW (Store Word) instruction with unaligned address
# Should report an error and not write to memory.
li x1, 0xDEAD # Data to attempt to store
li a0, 0x1002 # Base address: a0 = 0x1002
sw x1, -1(a0) # Attempt to store at mem[0x1002 - 1] = mem[0x1001] (unaligned)
ecall 0x3FF