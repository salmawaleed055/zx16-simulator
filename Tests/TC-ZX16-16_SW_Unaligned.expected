0x0000: 0271    li x1, 1
0x0002: 23B9    li a0, 4100
0x0004: F2A3    sw x1, -1(a0)
Store word address 0x1001 not word-aligned
0x0006: FFC7    ecall 0x3FF

--- Final State ---
x0: 0x0000
x1: 0x0001
x2: 0x0000
x3: 0x0000
x4: 0x0000
x5: 0x0000
a0: 0x1002
a1: 0x0000
PC: 0x0008
---------------------
Simulation finished.